src/implicit_function.jl

```diff
@@ -1,3 +1,21 @@
+struct Forward{returns_byproduct,F}
```

Comment:
The convention we used before was `return_byproduct` (no final `s`). It's not important except for consistency.

Reply:
I used `byproduct` across the whole code base as the type parameter. This is more verbose than BP but I think it makes it self-explanatory for people reading the src.